File: KataFizzBuzz/KataFizzBuzz/FizzBuzz.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace KataFizzBuzz
{
    public class FizzBuzz
    {
        public string Translate(int number)
        {
            if (number % 15 == 0)
                return "FizzBuzz";

            if (number % 5 == 0)
                return "Buzz";

            if (number % 3 == 0)
                return "Fizz";

            return number.ToString();
        }
    }
}

File: KataFizzBuzz.RSpec/fizzbuzz.rb
class FizzBuzz
	def translate(number)
		return "fizzbuzz" if number % 15 == 0
		return "buzz" if number % 5 == 0
		return "fizz" if number % 3 == 0

		return number.to_s
	end

	def translate_all(start, stop)
		(start..stop).collect do |number|
			translate number
		end
	end
end

